src/Couchbase.Lite.Shared/Query/QueryOrdering.cs

```diff
@@ -70,7 +71,7 @@ internal sealed class SortOrder : QueryOrderBy, ISortOrder
 
         #region Constructors
 
-        public SortOrder(IExpression expression) : base(null)
+        public SortOrder([NotNull]IExpression expression) : base(null)
         {
             IsAscending = true;
```

Comment:
Every `[NotNull]` should be combined with either `Debug.Assert` (internal) or something from `CBDebug` unless it immediately passes the parameter to something that does.